actors/verifreg/src/lib.rs

```diff
@@ -762,6 +811,133 @@ where
+
+    let mut allocation_total = StoragePower::zero();
+    for req in &reqs.requests {
+        allocation_total += StoragePower::from(req.size.0);
+
+        // Size must be at least the policy minimum.
+        if StoragePower::from(req.size.0) < policy.minimum_verified_allocation_size {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation size {} below minimum {}",
+                req.size.0,
+                policy.minimum_verified_allocation_size
+            ));
+        }
+
+        // Term must be at least the policy minimum.
+        if req.term_min < policy.minimum_verified_allocation_term {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term min {} below limit {}",
+                req.term_min,
+                policy.minimum_verified_allocation_term
+            ));
+        }
+        // Term cannot exceed the policy maximum.
+        if req.term_max > policy.maximum_verified_allocation_term {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term max {} above limit {}",
+                req.term_max,
+                policy.maximum_verified_allocation_term
+            ));
+        }
+        // Term range must be non-empty.
+        if req.term_min > req.term_max {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term min {} exceeds term max {}",
+                req.term_min,
+                req.term_max
+            ));
+        }
+
+        // Allocation must expire soon enough.
+        let max_expiration = curr_epoch + policy.maximum_verified_allocation_expiration;
+        if req.expiration > max_expiration {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation expiration {} exceeds maximum {}",
+                req.expiration,
+                max_expiration
+            ));
+        }
+    }
+
+    // Allocation size must match the tokens received exactly (we can't return change).
+    if &allocation_total != datacap_amount {
+        return Err(actor_error!(
+            illegal_argument,
+            "total allocation size {} must match data cap amount received {}",
+            allocation_total,
+            datacap_amount
+        ));
+    }
+
+    Ok(reqs)
+}
+
+// Checks that an address corresponsds to a miner actor.
+fn resolve_miner_id<BS, RT>(rt: &mut RT, addr: &Address) -> Result<ActorID, ActorError>
+where
+    BS: Blockstore,
+    RT: Runtime<BS>,
+{
+    let id = rt
+        .resolve_address(addr)
+        .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
+            format!("failed to resolve provider address {}", addr)
+        })?
+        .id()
+        .unwrap();
+    let code_cid =
+        rt.get_actor_code_cid(addr).with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
```

Comment:
If this was just a bad address wouldn't we already fail on resolve_address?  If so this should be ILLEGAL_STATE.

Reply:
I see what you mean, but it's not this actor's state that is busted. Basically the runtime is returning nonsensical results. I'll leave it because I don't think ILLEGAL_STATE is better, though I agree this isn't perfect.